Test/Unit/ThirdPartyModules/Mageplaza/ShippingRestrictionTest.php

```diff
@@ -0,0 +1,88 @@
+<?php
+/**
+ * Bolt magento2 plugin
+ *
+ * NOTICE OF LICENSE
+ *
+ * This source file is subject to the Open Software License (OSL 3.0)
+ * that is bundled with this package in the file LICENSE.txt.
+ * It is also available through the world-wide-web at this URL:
+ * http://opensource.org/licenses/osl-3.0.php
+ *
+ * @category   Bolt
+ * @package    Bolt_Boltpay
+ * @copyright  Copyright (c) 2020 Bolt Financial, Inc (https://www.bolt.com)
+ * @license    http://opensource.org/licenses/osl-3.0.php  Open Software License (OSL 3.0)
+ */
+
+namespace Bolt\Boltpay\Test\Unit\ThirdPartyModules\Mageplaza;
+
+use Bolt\Boltpay\ThirdPartyModules\Mageplaza\ShippingRestriction;
+use Magento\Framework\TestFramework\Unit\Helper\ObjectManager;
+use Magento\Quote\Model\Quote;
+use PHPUnit\Framework\TestCase;
+use Magento\Framework\Registry;
+
+/**
+ * Class ShippingRestrictionTest
+ * @package Bolt\Boltpay\Test\Unit\ThirdPartyModules\Mageplaza
+ * @coversDefaultClass \Bolt\Boltpay\ThirdPartyModules\Mageplaza\ShippingRestriction
+ */
+class ShippingRestrictionTest extends TestCase
+{
+    const QUOTE_ID = 1;
+
+    /** @var Registry */
+    private $coreRegistry;
+
+    /**
+     * @var Quote
+     */
+    private $quote;
+
+    /**
+     * @var Quote\Address
+     */
+    private $quoteAddress;
+
+    /**
+     * @var ShippingRestriction
+     */
+    private $currentMock;
+
+    /**
+     * @inheritdoc
+     */
+    public function setUp()
+    {
+        $this->coreRegistry = $this->quote = $this->createPartialMock(Registry::class, ['register']);
+        $this->quote = $this->createPartialMock(Quote::class, [
+            'getBoltParentQuoteId',
+            'getShippingAddress'
+        ]);
+        $this->quoteAddress = $this->createMock(Quote\Address::class);
+        $this->currentMock = (new ObjectManager($this))->getObject(
+            ShippingRestriction::class,
+            [
+                'coreRegistry' => $this->coreRegistry
+            ]
+        );
+    }
+
+    /**
+     * @test
+     * @covers ::afterLoadSession
+     */
+    public function afterLoadSession()
+    {
+        $this->quote->expects(self::once())->method('getBoltParentQuoteId')->willReturn(self::QUOTE_ID);
```

Comment:
I don't think we should have this type of tests.
In this test, we just check if **afterLoadSession** has all code it has now.
So if we need to refactor **afterLoadSession** we should rewrite the test.

In unit tests we should pass into method arguments, mock calls and check the result. In this case test should pass even after refactoring.

If we have no way to create a valuable test we shoudn't create tests at all.

As I mentioned in tech spec we need functional tests for all 3p plugins related code.

Let's wait for Matt's opinion here.

Reply:
@matt-thomason

Reply:
@vitaliyreznikov @johnmccombs1 I agree here - I think tests to just bump coverage are not very useful.

Reply:
> As I mentioned in tech spec we need functional tests for all 3p plugins related code.
How can we accomplish this?
I guess we need an instance per 3p module, with module deployed and configured, to run functional test against.

Reply:
Ideally we need to do it in circle ci instance:
- install module
- configure module (such as create gift cart code)
- run tests
- delete module
- install next module

Hopefully, we can start this during Q4